misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -255,6 +255,19 @@ private void markVisibleEp(VariableSymbol variableSymbol, JsonObject symbolJson,
                 updateVisibleEP(node, rawType, false);
             }
         }
+        if (rawType.typeKind() == TypeDescKind.UNION) {
+            UnionTypeSymbol unionTypeSymbol = (UnionTypeSymbol) rawType;
+            unionTypeSymbol.memberTypeDescriptors().forEach(member -> {
+                TypeSymbol memberRawType = getRawType(member);
+                if (memberRawType.typeKind() == TypeDescKind.OBJECT) {
```

Comment:
Shall we merge if conditions?

Reply:
Thanks. Refactor the source with https://github.com/ballerina-platform/ballerina-lang/pull/36125/commits/907c5d622963ecbf6d42e15b167330f6c4b36861 commit.